File: code/examples/Demo-app/android/DemoApp/src/main/kotlin/com/opencloudsheet/demo/model/Expense.kt
package com.opencloudsheet.demo.model

import com.opencloudsheet.model.worksheet.IWorksheetRow

/**
 * Expense model for tracking expenses in a workbook.
 *
 * Fields are mapped to worksheet columns in declaration order:
 * - Column 0 (rowId) - Auto-managed UUID
 * - Column 1 (createdAt) - Auto-managed timestamp
 * - Column 2 (updatedAt) - Auto-managed timestamp
 * - Column 3 (name) - Expense name/description
 * - Column 4 (amount) - Expense amount
 * - Column 5 (currency) - Currency code (e.g., USD, EUR, INR)
 */
data class Expense(
    var name: String = "",
    var amount: Double = 0.0,
    var currency: String = "USD"
) : IWorksheetRow() {
    override fun getIosClassName() = "CloudSheetDemo.Expense"
    override fun getTableColumnFieldsOrder() = listOf("name", "amount", "currency")
}

File: code/examples/Demo-app/ios/CloudSheetDemo/CloudSheetDemo/Models/Expense.swift
//
//  Expense.swift
//  CloudSheetDemo
//
//  Expense model for tracking expenses in a workbook
//

import Foundation
import OpenCloudSheet

struct Expense: IWorksheetRow {
    public var _rowId: String = ""
    public var _createdAt: Int64 = 0
    public var _updatedAt: Int64 = 0
    public var _rowIndex: Int = -1

    var name: String
    var amount: Double
    var currency: String

    static func getAndroidClassName() -> String {
        return "com.opencloudsheet.demo.model.Expense"
    }

    static func getTableColumnFieldsOrder() -> [String] {
        return ["name", "amount", "currency"]
    }

    init(name: String, amount: Double, currency: String) {
        self.name = name
        self.amount = amount
        self.currency = currency
    }
}
